docs/design.md

```diff
@@ -0,0 +1,146 @@
+A separate app for collecting sensor data that is needed to estimate physical capacity. 
+* The Niceday app itself cannot gather this data.
+* At the start of an activity the sensor data collector is triggered to start recording sensor data.
+* After an activity or upon Wifi connection the data is pushed to the PerfectFit sensor data REST API.
+* Will be based on an existing app and developed outside of scope of PerfectFit
+virtual coach system.
+--- 
+
+### Niceday components
+#### Niceday server (already developed)
+* Backend for the Niceday app.
+* Data storage: 
+  - user profile data, 
+  - app-interaction data
+  - messages
+* Handles authentication
+
+--- 
+### PerfectFit virtual coach system components
+We cannot change much in the NiceDay app, new PerfectFit features will be implemented 
+through the virtual coach that interacts with the user through the messaging function of the 
+NiceDay app. We will make use of Niceday trackers to gather data, 
+and make use of the Niceday daily planner to plan activities.
+
+
+### Sensor data REST API
+A Sensor REST API that allows for the sensor data collection app to send data to PerfectFit system.
+* Validates and preprocesses data
+* Writes data directly to the database using the ORM
+
+### Rasa agent
+The conversational agent developed in the [Rasa](https://rasa.com/) framework.
+* Given an input message of the user, perform the appropriate action.
+* Actions can range from a simple text message response to something more complex, using any combination of:
+  - using an algorithm (through calling functions from imported libraries, often with data from db or niceday) 
+  - querying the perfectfit database using ORM
+  - calling the niceday-api for information from niceday-server
+
+### Rasa agent REST API
+Allows for other components to interact with the agent.
+
+### niceday-broker
+A worker that sends back and forth messages from a niceday user between the Niceday server and the rasa agent.
+* Depends on the javascript Niceday client [goalie-js](https://github.com/senseobservationsystems/goalie-js).
+* Listens to incoming messages, upon an incoming message it sends a request to the rasa agent REST API
+* It forwards any text response from rasa back to the user through Niceday
+
+### niceday-api
+Allows control over certain functionalities in the niceday app.
+It is a node.js REST API that wraps functions that we need from the javascript Niceday client 
+[goalie-js](https://github.com/senseobservationsystems/goalie-js).
+* Request data from the niceday server
+* Read from and manage Niceday (custom) trackers
+* Interact with Niceday daily planner
+
+### PostgreSQL database
+Stores PerfectFit-specific data, 
+* Stores:
+  - Data about the user (name, age, stage of patient journey, preferences)
+  - Sensor data
+  - Any data that we need to conduct research
+* The interface to it will be an SqlAlchemy ORM (so no API abstraction)
+
+### Content Management System
```

Comment:
Nice to the abbreviation `CMS` after the full name, which is used in the text below.

Reply:
Fixed :)